client/state/data-layer/wpcom/sites/alerts/fix-status.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'calypso/state/data-layer/wpcom-http/utils';
+import { JETPACK_SCAN_THREATS_GET_FIX_STATUS } from 'calypso/state/action-types';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import { getFixThreatsStatus } from 'calypso/state/jetpack-scan/threats/actions';
+import { requestScanStatus } from 'calypso/state/jetpack-scan/actions';
+import { requestJetpackScanHistory } from 'calypso/state/jetpack-scan/history/actions';
+import { registerHandlers } from 'calypso/state/data-layer/handler-registry';
+import { http } from 'calypso/state/data-layer/wpcom-http/actions';
+
+const POOL_EVERY_MILLISECONDS = 1000;
+
+export const request = ( action ) => {
+	return [
+		http(
+			{
+				apiNamespace: 'wpcom/v2',
+				method: 'GET',
+				path: `/sites/${ action.siteId }/alerts/fix/`,
+				query: { threat_ids: action.threatIds },
+			},
+			{ ...action }
+		),
+	];
+};
+
+export const success = ( action, fixer_state ) => {
+	const threatArray = Object.values( fixer_state.threats );
+
+	const inProgressThreats = threatArray.filter( ( threat ) => threat.status === 'in_progress' );
+
+	if ( inProgressThreats.length > 0 ) {
+		return [
+			( dispatch ) =>
+				setTimeout(
+					() => dispatch( getFixThreatsStatus( action.siteId, action.threatIds ) ),
+					POOL_EVERY_MILLISECONDS
+				),
+		];
+	}
+
+	const fixedThreats = threatArray.filter( ( threat ) => threat.status === 'fixed' );
+
+	if ( fixedThreats.length === action.threatIds.length ) {
+		return [
+			successNotice( i18n.translate( 'All selected threats were successfully fixed.' ), {
+				duration: 4000,
+			} ),
+			requestScanStatus( action.siteId ),
+			// Since we can fix threats from the History section, we need to update that
+			// information as well.
+			requestJetpackScanHistory( action.siteId ),
+		];
+	}
+
+	return [
+		errorNotice(
+			i18n.translate( 'Not all selected threats could be fixed. Please contact our support.' ),
```

Comment:
Does it make sense to talk about "selected threats"? As I understand it, the user can fix either one threat, or all threats?

Reply:
> Please contact our support

"Please contact support" is in more common use, I think. Should it link to the Jetpack support page?

Reply:
I was trying to come up with a generic phrase that could fit in both cases. But perhaps it would be better to check if it's a single threat being fixed and adjust the message accordingly.